### PR TITLE
integration: Add further native asset atomic swap integration tests

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
This PR adds further integration testing for native asset support in atomic settlement.

### Testing
- All integration tests pass

### CODE REVIEW

File: contracts-stylus/src/contracts/core/core_settlement.rs [hunk 1]
```diff
@@ -277,6 +278,16 @@ impl CoreSettlementContract {
...
 
+        // The transaction value should be zero unless the external party is selling
+        // native ETH in the trade
+        let match_result = &valid_match_settle_atomic_statement.match_result;
+        let is_native_eth = is_native_eth_address(match_result.base_mint);
+        let is_external_party_sell = match_result.is_external_party_sell();
+        let native_eth_sell = is_native_eth && is_external_party_sell;
+        if !native_eth_sell && msg::value() > U256::ZERO {
+            return Err(INVALID_TRANSACTION_VALUE_ERROR_MESSAGE.into());
+        }
```
Comment (akirillo) [inline contracts-stylus/src/contracts/core/core_settlement.rs:289]: this is a great check to have, nice catch

File: integration/src/tests.rs [hunk 2]
```diff
@@ -1443,7 +1443,8 @@ async fn test_process_atomic_match_settle__native_asset_sell_side(ctx: TestConte
     let protocol_balance = ctx.get_erc20_balance_of(quote_addr, protocol_fee_addr).await?;
 
     let expected_quote_balance = initial_quote_balance + match_result.quote_amount - fees.total();
-    let expected_eth_balance = initial_eth_balance - match_result.base_amount - eth_gas_cost;
+    let expected_eth_balance =
+        initial_eth_balance - eth_gas_cost + match_result.base_amount - fees.total();
```
Comment (akirillo) [inline integration/src/tests.rs]: looks like you replicated the buy-side expected base balance calculation here, this change should be undone

File: integration/src/tests.rs [hunk 3]
```diff
@@ -1513,9 +1580,58 @@ async fn test_process_atomic_match_settle_inconsistent_protocol_fee(
...
 integration_test_async!(test_process_atomic_match_settle_inconsistent_protocol_fee);
 
+/// Test `process_atomic_match_settle` with invalid transaction value
+async fn test_process_atomic_match_settle_invalid_transaction_value(
+    ctx: TestContext,
+) -> Result<()> {
+    let contract = ctx.darkpool_contract();
+    let data = setup_atomic_match_settle_test_native_eth(false /* buy_side */, &ctx).await?;
+
+    // Try to execute with insufficient ETH value (half of required)
```
Comment (akirillo) [inline integration/src/tests.rs]: nit
```suggestion
    // Try to execute with insufficient ETH value
```

**Review by akirillo — APPROVED** [review]

File: integration/src/tests.rs (hunk 2 above)
Comment (joeykraut) [reply integration/src/tests.rs]: oops, left that in while i was making the test fail

**Pull request merged**